pepper-apis/studybuilder-cli/tenants/register-user-in-pepper-action.js

```diff
@@ -0,0 +1,389 @@
+       console.log('setting userGUID claim from temp user');
+       api.idToken.setCustomClaim(pepperUserGuidClaim, pepper_params.tempUserGuid);
+       api.user.setAppMetadata("user_guid", pepper_params.tempUserGuid);
+       } else if (event.user.app_metadata.user_guid) {
+       api.idToken.setCustomClaim(pepperUserGuidClaim, event.user.app_metadata.user_guid);
+       } */
+
+      var pepperUrl = event.secrets.pepperBaseUrl;
+      if (event.client.metadata.backendUrl) {
+        pepperUrl = event.client.metadata.backendUrl;
+      }
+      var request = require('request');
+      console.log('Invoking DDP registration : ' + pepperUrl + ' ... params: ' + JSON.stringify(pepper_params));
+      console.log('ID Token:' + JSON.stringify(api.idToken));
+      request.post({
+        url: pepperUrl + '/pepper/v1/register',
+        json: pepper_params,
+        timeout: 15000
+      }, function (err, response, body) {
+        if (err) {
+          console.log('Error while registering auth0 user ' + event.user.user_id);
+          console.log(err);
+          let error = new Error(JSON.stringify({
+            code: err.code,
+            message: err.message,
+            statusCode: 500
+          }));
+          console.log('Error during registration: ' + err.message);
+          return api.access.deny(err.message);
+        } else if (response && response.statusCode !== 200) {
+          console.log('Response: ' + JSON.stringify(response));
+          console.log('Failed to register auth0 user ' + event.user.user_id + ':' + response.statusCode + ' at ' + pepperUrl);
+          console.log('BODY: ' + JSON.stringify(body));
+          body = body || {};
+          let error = new Error(JSON.stringify({
+            code: body.code,
+            message: body.message,
+            statusCode: response.statusCode
+          }));
+
+          console.log('Access denied...' + body.message);
+          return api.access.deny(body.message);
+          //todo .. doesnt seem to block UI invoking next steps ?
+          //try redirect ?
+        } else {
+          console.log(' register response: ' + JSON.stringify(response));
+          console.log(' register body: ' + JSON.stringify(body));
+
+          // all is well
+          //todo.. below doesnt seem to work.. looks AppMetadata update is likely delayed
+          // added before register call for now
+          //maybe try API call?
+          var ddpUserGuid = body.ddpUserGuid;
+          api.idToken.setCustomClaim(pepperUserGuidClaim, ddpUserGuid);
+          api.user.setAppMetadata("user_guid", ddpUserGuid);
+          api.user.setUserMetadata("user_guid", ddpUserGuid);
+          console.log('updated user metaData with user GUID ? ');
+          api.user.setAppMetadata("user_guid_test_2", "test GUID-2");
+          api.user.setUserMetadata("user_guid_test_3", "test GUID-3");
+
+          // Update user AppMetadata with user_guid using Auth0 Management API
```

Comment:
Below API call code is a attempt to update user GUID in auth0 user AppMetadata which isnt working too and might be removed.